src/test/java/org/jboss/windup/rules/tests/WindupRulesMultipleTests.java

```diff
@@ -0,0 +1,526 @@
+        if (StringUtils.isBlank(testToExecute))
+        {
+            testToExecute = "";
+        }
+        Pattern testToExecutePattern = Pattern.compile(testToExecute);
+        FileSuffixPredicate predicate = null;
+        if (findingTestFiles)
+        {
+            predicate = new FileSuffixPredicate("\\.(windup|rhamt)\\.test\\.xml");
+        }
+        else
+        {
+            predicate = new FileSuffixPredicate("\\.(windup|rhamt)\\.xml");
+        }
+        final File directory = new File("rules");
+        final File rulesReviewed = new File("rules-reviewed");
+        final List<File[]> rulesetToTest = new ArrayList<>();
+        FileVisit.visit(directory, predicate).stream()
+                    .filter(file -> testToExecutePattern.matcher(file.toString()).find()).map(file -> new File[] { file, directory })
+                    .forEach(rulesetToTest::add);
+        FileVisit.visit(rulesReviewed, predicate).stream()
+                    .filter(file -> testToExecutePattern.matcher(file.toString()).find())
+                    .map(file -> new File[] { file, rulesReviewed }).forEach(rulesetToTest::add);
+
+        return rulesetToTest;
+    }
+
+    @Deployment
+    @AddonDependencies({
+            @AddonDependency(name = "org.jboss.windup.exec:windup-exec"),
+            @AddonDependency(name = "org.jboss.windup.config:windup-config-xml"),
+            @AddonDependency(name = "org.jboss.windup.rules.apps:windup-rules-java"),
+            @AddonDependency(name = "org.jboss.windup.rules.apps:windup-rules-java-ee"),
+            @AddonDependency(name = "org.jboss.windup.rules.apps:windup-rules-java-project"),
+            @AddonDependency(name = "org.jboss.windup.rules.apps:windup-rules-xml"),
+            @AddonDependency(name = "org.jboss.windup.reporting:windup-reporting"),
+            @AddonDependency(name = "org.jboss.windup.utils:windup-utils"),
+            @AddonDependency(name = "org.jboss.forge.furnace.container:cdi")
+    })
+    public static AddonArchive getDeployment()
+    {
+        return ShrinkWrap.create(AddonArchive.class)
+                .addBeansXML()
+                .addPackage(WindupRulesMultipleTests.class.getPackage());
+    }
+
+    @Inject
+    private Furnace furnace;
+
+    @Inject
+    private GraphContextFactory factory;
+
+    @Inject
+    private WindupProcessor processor;
+    
+    @Inject
+    private TestRulesPath testRulesPath;
+
+    @Inject
+    private RulesPath rulesPath;
+
+    
+    @Before
+    public void before()
+    {
+        testRulesPath.setRules(data(true));
+        rulesPath.setRules(data(false));
+    }
+
+    @Test
+    public void executeRule()
+    {
+        try {
+            testRulesPath.resetIterator();
+
+            File[] ruleFiles = rulesPath.getNextRule();
+            File[] ruleTestFiles = findMatchingTestFile(ruleFiles[0], ruleFiles[1]);
+            Assert.assertNotNull("No test file found", ruleTestFiles);
```

Comment:
The actual test is done within the other test method, so the test to do with presence or absence of test files and test rule ids is atomic. This is purely here to prevent a null error when we run the execution tests from this method.

Reply:
So we could just have the same error message (that it's already in the [documentation PR](https://github.com/windup/windup-documentation/pull/206)) we have in https://github.com/windup/windup-rulesets/pull/335/files#diff-5e70928c242f2bc815464279d075d8dcR449 also here: "No test file matching rule" instead of "No test file found"